app/models/journal_manager.rb

```diff
@@ -225,28 +229,70 @@ def self.recreate_initial_journal(type, journal, changed_data)
   end
 
   def self.add_journal!(journable, user = User.current, notes = '')
-    if journalized? journable
-      # Obtain a table lock to ensure consistent version numbers
-      Journal.with_write_lock(journable) do
+    return unless journalized?(journable)
+
+    # Ensure a version exists for this journable type
+    # since no version is changed here, in case of concurrency, one
+    # of the calls is allowed to fail
+    journable_type = base_class_name(journable.class)
+    ::JournalVersion.find_or_create_by(journable_type: journable_type, journable_id: journable.id)
+
+    # Increment the version according to our DB
+    version =
```

Comment:
This would look good in a method, too. I know we'll remove the branch in dev but it still makes sense for 9.x.